src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -751,7 +751,19 @@ internal static bool IsPackageReferenceProject(Project project)
                         {
                             try
                             { // In case proj and assets file are not in sync and some refs were deleted
-                                installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                var isPackageCentrallyManaged = tfmInformation.CentralPackageVersions.Any(cpv => cpv.Key.Equals(topLevelPackage.Name, StringComparison.Ordinal));
+                                if (isPackageCentrallyManaged)
+                                {
+                                    var packageCentralVersion = tfmInformation.CentralPackageVersions.Where(cpv => cpv.Key.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
+                                    installedPackage = new InstalledPackageReference(topLevelPackage.Name)
+                                    {
+                                        OriginalRequestedVersion = packageCentralVersion.Value.VersionRange.MinVersion.ToString(),
+                                    };
+                                }
+                                else
+                                {
+                                    installedPackage = projPackages.Where(p => p.Name.Equals(topLevelPackage.Name, StringComparison.Ordinal)).First();
```

Comment:
Does `VersionOverride` work today?

Reply:
Yess it works for CPM, in the Samples repo we have an example I have been using to test this

Reply:
Does `dotnet list package` work with it? Does it display the correct version?

Reply:
This is an example, the version in Directory.Packages.props is 13.0.0 and the VersionOverride is 9.0.1
![image](https://github.com/NuGet/NuGet.Client/assets/43253759/ca9802a0-b95d-44e9-bf0c-7a27849c4c5c)

Reply:
That's incorrect in that case. We need to have requested be `9.0.1`.

Reply:
I see your point and I believe its valid. I was looking at it the other way. In this example: For me I know that I'm requesting 13.0.0 but I want that my `VersionOverride` version is the one that gets Resolved. 

@jeffkl Can you help us clarify the expected behavior here?

Reply:
VersionOverride changes what you're requesting. 
It's basically overriding what the PackageVersion wanted.

Reply:
Thanks for the clarification! I'll change the behavior of that

Reply:
Maybe we can add a visual indicator that there's a version override?  
Similar to say "Auto Referenced packages". 

I'm just brainstorming here, you can wait for Jeff to confirm :)

Reply:
Package ID is case insensitive, so this comparer needs updated.